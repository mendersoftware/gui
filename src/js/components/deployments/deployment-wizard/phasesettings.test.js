import React from 'react';
import renderer from 'react-test-renderer';
import PhaseSettings from './phasesettings';

let dateMock;

describe('PhaseSettings Component', () => {
  beforeEach(() => {
    const mockDate = new Date('2019-01-01T13:00:00.000Z');
    const _Date = Date;
    global.Date = jest.fn(() => mockDate);
    global.Date.parse = _Date.parse;
    global.Date.now = _Date.now;
    global.Date.toISOString = _Date.toISOString;
    global.Date.UTC = _Date.UTC;
    dateMock = jest.spyOn(global, 'Date').mockImplementation(() => mockDate);
  });

  afterEach(() => {
    dateMock.mockRestore();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<PhaseSettings phases={[{ batch_size: 0 }]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
