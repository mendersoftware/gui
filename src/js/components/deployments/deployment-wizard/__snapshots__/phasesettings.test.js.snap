// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseSettings Component renders correctly 1`] = `
<div>
  <table
    className="MuiTable-root"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        />
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Batch size
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Phase begins
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        >
          Delay before next phase
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
          scope="col"
        />
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          scope="row"
        >
          <div
            className="MuiChip-root MuiChip-sizeSmall"
            onKeyUp={[Function]}
          >
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              Phase 1
            </span>
          </div>
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          100
          <span
            className="info"
            style={
              Object {
                "marginLeft": "5px",
              }
            }
          >
            (NaN devices)
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          <time
            dateTime="2019-01-01T13:00:00+00:00"
            title={null}
          >
            2019-01-01 13:00
          </time>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        >
          -
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
        />
      </tr>
    </tbody>
  </table>
  <div
    className="MuiButtonBase-root MuiChip-root margin-top-small MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-clickable"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiChip-icon MuiChip-iconColorPrimary"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
    <span
      className="MuiChip-label"
    >
      Add a phase
    </span>
  </div>
</div>
`;
